moveos/moveos-types/src/state_resolver.rs

```diff
@@ -75,17 +76,30 @@ where
                 );
                 Ok(s.value)
             })
-            .transpose()
+            .transpose();
+
+        match result {
+            Ok(opt) => {
+                if let Some(data) = opt {
+                    Ok((Some(data), 0))
+                } else {
+                    Ok((None, 0))
+                }
+            }
+            Err(err) => Err(err),
+        }
     }
 }
 
 impl<R> ModuleResolver for MoveOSResolverProxy<R>
 where
     R: StateResolver,
 {
-    type Error = anyhow::Error;
+    fn get_module_metadata(&self, _module_id: &ModuleId) -> Vec<Metadata> {
+        vec![]
```

Comment:
Just return an empty array? Do we use the `get_module_metadata` in the VM?

Reply:
Currently, the metadata for the module is manually read from the VM and the function is not invoked.